dvc/api.py

```diff
@@ -25,12 +24,20 @@ def get_url(path, repo=None, rev=None, remote=None):
         with reraise(FileNotFoundError, PathMissingError(path, repo)):
             metadata = _repo.repo_fs.metadata(path_info)
 
-        if not metadata.is_dvc:
+        if (
+            not metadata.is_dvc
+            or len(metadata.outs) != 1
+            or not metadata.outs[0].hash_info
+        ):
             raise OutputNotFoundError(path, repo)
 
         cloud = metadata.repo.cloud
-        hash_info = get_hash(path_info, _repo.repo_fs, "md5", _repo.odb)
```

Comment:
This was actively trying to compute a hash, which is clearly not something that `get_url` should do. What we really needed is a recorded md5, which is now easy to get from `dvcfs.info()`(part of fsspec).